include/dlaf/mc/cholesky.h

```diff
@@ -0,0 +1,306 @@
+          if (row_comm_size > 1) {
+            // Receiving the panel
+            panel[i_local] = hpx::dataflow(  //
+                hpx::util::unwrapping(
+                    [](auto index, auto&& tile_size, auto&& comm_wrapper) -> Tile<const T, Device::CPU> {
+                      memory::MemoryView<T, Device::CPU> mem_view(
+                          util::size_t::mul(tile_size.rows(), tile_size.cols()));
+                      Tile<T, Device::CPU> tile(tile_size, std::move(mem_view), tile_size.rows());
+                      dlaf::comm::sync::broadcast::receive_from(index, comm_wrapper().rowCommunicator(),
+                                                                tile);
+                      return std::move(tile);
+                    }),
+                k_rank_col, mat.tileSize(GlobalTileIndex(i, k)), serial_comm());
+          }
+        }
+      }
+
+      auto nextlocaltilek = distr.nextLocalTileFromGlobalTile<Coord::Col>(k + 1);
+      for (SizeType j_local = nextlocaltilek; j_local < localnrtile_cols; ++j_local) {
+        auto j = distr.globalTileFromLocalTile<Coord::Col>(j_local);
+
+        // Choose "high priority" for first tile of the trailing matrix
+        // Version 1: only the global k + 1 column is "high priority"
+        auto trailing_matrix_executor = (j == k + 1) ? matrix_HP_executor : matrix_normal_executor;
+
+        // TODO: benchmark to find faster
+        // Version 2: the next column of each rank is "high priority"
+        //        auto trailing_matrix_executor = (j_local == nextlocaltilek) ? matrix_HP_executor :
+        //        matrix_normal_executor;
+
+        hpx::shared_future<Tile<const T, Device::CPU>> col_panel;
+
+        auto j_rank_row = distr.rankGlobalTile<Coord::Row>(j);
+
+        if (mat.rankIndex().row() == j_rank_row) {
+          auto i_local = distr.localTileFromGlobalTile<Coord::Row>(j);
+
+          // Avoid useless communications if one-row communicator grid and if on the last panel
+          if (col_comm_size > 1 && j != (mat.size().cols() - 1)) {
+            // Broadcast the (trailing) panel column-wise
+            hpx::dataflow(hpx::util::unwrapping([k, j](auto&& tile, auto&& comm_wrapper) {
+                            dlaf::comm::sync::broadcast::send(comm_wrapper().colCommunicator(), tile);
+                          }),
+                          panel[i_local], serial_comm());
+          }
+
+          // Check if the diagonal tile of the trailing matrix is on this node and
+          // compute first tile of the column of the trailing matrix: diagonal element mat(j,j), reading
+          // mat.read(j,k), using herk (blas operation)
+          hpx::dataflow(trailing_matrix_executor, hpx::util::unwrapping(tile::herk<T, Device::CPU>),
+                        uplo, blas::Op::NoTrans, -1.0, panel[i_local], 1.0,
+                        mat(LocalTileIndex{i_local, j_local}));
+
+          col_panel = panel[i_local];
```

Comment:
I think you can return the tile directly from the "broadcast" dataflow.

Reply:
@albestro Why do you want to add an extra dependency on the broadcast?

Reply:
You're right. My mistake.

It was misleading for me, I would reorder the code blocks in a few points...but it's ok!